### PR TITLE
PoW Faucet

### PR DESCRIPTION
**Pull request by harryttd**
Tezos faucet chart updates for the new PoW challenge functionality.

See https://github.com/oxheadalpha/tezos-faucet-backend/pull/2 and https://github.com/oxheadalpha/tezos-faucet/pull/16

Notable changes:
- Can configure separate RPC url for the backend code. This way the backend can query the k8s node rpc service directly.
- Deploying the frontend is now optional, so you can have just a programmatic faucet.
- Added pod security context, service account, plus other Helm k8s configuration options.
- Removed unused secret.
- General cleanup and improvement.

### CODE REVIEW

File: charts/tezos-faucet/values.yaml [hunk 1]
```diff
@@ -1,33 +1,72 @@
...
-  tezos_faucet_backend: ghcr.io/oxheadalpha/tezos-faucet-backend:latest
-authorizedHost: "'*'"
+# Faucet backend: https://github.com/oxheadalpha/tezos-faucet-backend
+# Faucet frontend: https://github.com/oxheadalpha/tezos-faucet
 
-# if your balance is over the max, the faucet won't give you more tokens
-maxBalance: 12000
+images:
+  tezosFaucet: ghcr.io/oxheadalpha/tezos-faucet:2.0.0
+  tezosFaucetBackend: ghcr.io/oxheadalpha/tezos-faucet-backend:2.0.0
```
Comment (nicolasochem) [inline charts/tezos-faucet/values.yaml:6]: why not latest?

File: charts/tezos-faucet/templates/deployment.yaml [hunk 2]
```diff
@@ -15,34 +15,77 @@ spec:
...
         # https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments
-        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }} 
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
     spec:
+      restartPolicy: Always
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "tezos-faucet.serviceAccountName" . }}
```
Comment (nicolasochem) [inline charts/tezos-faucet/templates/deployment.yaml:25]: why is this needed?

File: charts/tezos-faucet/templates/configmap.yaml [hunk 3]
```diff
@@ -1,21 +1,32 @@
...
 apiVersion: v1
-data:
-  config.json: |
-{{ .Values.config | mustToPrettyJson | indent 4 }}
 kind: ConfigMap
 metadata:
   name: faucet-config
   namespace: {{ .Release.Namespace }}
+data:
+  config.json: |
```
Comment (nicolasochem) [inline charts/tezos-faucet/templates/configmap.yaml:8]: why replace a kitchen-sink configuration passthrough with something more rigid?

File: charts/tezos-faucet/values.yaml (hunk 1 above)
Comment (harryttd) [reply charts/tezos-faucet/values.yaml:6]: If faucet code changes and releases, then  it's possible a faucet deployment could end up pulling the latest images, while the chart code is not updated to handle it.

File: charts/tezos-faucet/templates/deployment.yaml (hunk 2 above)
Comment (harryttd) [reply charts/tezos-faucet/templates/deployment.yaml:25]: it's just one of the properties that exist on sts's and deployments. Chart generator will create one with it. It's so the pods can interact as a specific service account that may have custom permissions.

File: charts/tezos-faucet/templates/configmap.yaml (hunk 3 above)
Comment (harryttd) [reply charts/tezos-faucet/templates/configmap.yaml:8]: Profiles now have added configuration which is for the challenges. I can't just pass the profiles object like it currently is because it contains that info. If i did, the frontend would be getting that info and it shouldn't. The frontend should not know anything about the difficulty and amount of challenges needed. Albeit right now these numbers are not dynamic, they will be more dynamic as we go forward.

**Review by nicolasochem — APPROVED** [review]
lgtm
